spec/updater_spec.rb

```diff
@@ -7,77 +7,84 @@
 require 'validator'
 
 RSpec.describe Updater do
-  let(:database) { ArrayDatabase.new }
-  let(:input) { StringIO.new(quick_exit_responces) }
   let(:output) { StringIO.new }
   let(:quick_exit_responces) { "0\nname\nirrelevant\nn\nn\n" }
-  let(:user_interface) { UserInterface.new(input, output, validator) }
-  let(:validator) { Validator.new }
 
   describe '#run' do
-    before(:each) { database.create(test_details) }
-
-    it 'asks the user for a contact to update' do
+    it 'displays that no contacts are found if database is empty' do
+      user_interface = UserInterface.new(StringIO.new("\n"), output, Validator.new)
+      database = ArrayDatabase.new
       updater = described_class.new(user_interface, database)
 
       updater.run
```

Comment:
why not use this:

```suggestion
      run_updater_with_empty_database
      
      #...
      
    def run_updater_with_empty_database
      user_interface = UserInterface.new(StringIO.new("\n"), output, Validator.new)
      database = ArrayDatabase.new

      described_class.new(user_interface, database).run
    end
```

and you can reuse it in the next test

Reply:
because it only happens twice, you lead me to belive that in TDD 3 is the magic number

Reply:
yeah it is in most cases, but this is what I told you the other day about those numbers being a guidance rather than a dogma. At the end of the day is your criteria as a developer to judge what's best. 

Although you would usually wait until you have three cases, in this particular case, it looked like it would be helpful and made sense, even though there were only two cases.